src/main/java/Duke.java

```diff
@@ -1,10 +1,166 @@
+                    String datelineDateBy = content.split("/")[1].split(" ")[1];
+                    taskList.add(new Deadline(deadlineTask,datelineDateBy ));
+                    addTaskSuccessMessage(taskList, "\tGot it. I've added this deadline: ");
+                    break;
+                    
+                case "event":
+                    System.out.println("event");
+                    String eventTask = content.split("/")[0].toLowerCase();
+                    String eventDateBy = content.split("/")[1];
+                    eventDateBy = eventDateBy.substring(eventDateBy.indexOf(" "));
+                    taskList.add(new Event(eventTask,eventDateBy));
+                    addTaskSuccessMessage(taskList, "\tGot it. I've added this Event: ");
+                    break;
+
+                case "list":
+
+                    String listReturnString = String.format("%s%s%s",lineDivider,getList(taskList),lineDivider);
+                    System.out.println(listReturnString);
+
+                    break;
+
+                case "done":
+                    markTaskDone(taskList, userInput, content);
+
+                    break;
+
+                case "bye":
+                    System.out.println(dukeFarewell);
+                    System.exit(0);
+                default:
+                    System.out.println("Command word not recognised - please start command with " +
+                            "'todo', 'deadline' or 'event'");
+            }
+        }
+
+    }
+
+    public static void markTaskDone(ArrayList<Task> taskList, String userInput, String content) {
+        if (userInput.matches(".*\\d.*")) { // checks if there is a number in done cmd
+            int taskNumber = Integer.parseInt(content) ;
+            int indexOfTaskToBeMarked = taskNumber - 1;
+            if (indexOfTaskToBeMarked < taskList.size()){
+                taskList.get(indexOfTaskToBeMarked).setDone(true);
+                String markedTaskAsDoneMessage =
+                        String.format("%s\t Nice! I've marked this task as done:\n %s %s \n%s \n",
+                                lineDivider,
+                                taskList.get(indexOfTaskToBeMarked).getStatusIcon(),
+                                taskList.get(indexOfTaskToBeMarked).getDescription(),
+                                lineDivider);
+                System.out.println(markedTaskAsDoneMessage);
+                } else {
+                String taskDoesNotExistMessage = String.format("%s\t Task does not exist!\n %s \n", lineDivider, lineDivider);
+                System.out.print(taskDoesNotExistMessage);
+            }
+        } else {
+            String doneErrorPrompt = "Which task do you want to mark done?";
+            String doneListMessage = String.format("%s\t%s\n\t%s\n%s",lineDivider,doneErrorPrompt,getList(taskList),lineDivider);
+            System.out.println(doneListMessage);
+        }
```

Comment:
i like how you have included error handlings even though it has not yet been specified in the weekly project requirements